pkg/apis/healthchecking/v1alpha1/machinedisruptionbudget_types.go

```diff
@@ -0,0 +1,87 @@
+
+	// Label query over machines whose deletions are managed by the disruption
+	// budget.
+	// +optional
+	Selector *metav1.LabelSelector `json:"selector,omitempty" protobuf:"bytes,2,opt,name=selector"`
+
+	// An deletion is allowed if at most "maxUnavailable" machines selected by
+	// "selector" are unavailable after the deletion.
+	// For example, one can prevent all voluntary deletions by specifying 0.
+	// This is a mutually exclusive setting with "minAvailable".
+	// +optional
+	MaxUnavailable *int32 `json:"maxUnavailable,omitempty" protobuf:"bytes,3,opt,name=maxUnavailable"`
+}
+
+// MachineDisruptionBudgetStatus represents information about the status of a
+// MachineDisruptionBudget. Status may trail the actual state of a system.
+type MachineDisruptionBudgetStatus struct {
+	// Most recent generation observed when updating this MDB status. MachineDisruptionsAllowed and other
+	// status information is valid only if observedGeneration equals to MDB's object generation.
+	// +optional
+	ObservedGeneration int64 `json:"observedGeneration,omitempty" protobuf:"varint,1,opt,name=observedGeneration"`
+
+	// DisruptedMachines contains information about machines whose deletion was
+	// processed by the API server but has not yet been observed by the MachineDisruptionBudget controller.
+	// A machine will be in this map from the time when the API server processed the
+	// deletion request to the time when the machine is seen by MDB controller
+	// as having been marked for deletion (or after a timeout). The key in the map is the name of the machine
+	// and the value is the time when the API server processed the deletion request. If
+	// the deletion didn't occur and a machine is still there it will be removed from
+	// the list automatically by MachineDisruptionBudget controller after some time.
+	// If everything goes smooth this map should be empty for the most of the time.
+	// Large number of entries in the map may indicate problems with machines deletions.
+	// +optional
+	DisruptedMachines map[string]metav1.Time `json:"DisruptedMachines,omitempty" protobuf:"bytes,2,rep,name=DisruptedMachines"`
+
+	// Number of machines disruptions that are currently allowed.
+	MachineDisruptionsAllowed int32 `json:"disruptionsAllowed" protobuf:"varint,3,opt,name=disruptionsAllowed"`
+
+	// current number of healthy machines
+	CurrentHealthy int32 `json:"currentHealthy" protobuf:"varint,4,opt,name=currentHealthy"`
+
+	// minimum desired number of healthy machines
+	DesiredHealthy int32 `json:"desiredHealthy" protobuf:"varint,5,opt,name=desiredHealthy"`
+
+	// total number of machines counted by this disruption budget
+	ExpectedMachines int32 `json:"ExpectedMachines" protobuf:"varint,6,opt,name=ExpectedMachines"`
+}
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// MachineDisruptionBudget is an object to define the max disruption that can be caused to a collection of machines
```

Comment:
nit: `s/ that can be caused to a collection of machines/ that a collection of machines can experience` sounds a bit better

Reply:
done